src/app/app.component.ts

```diff
@@ -1,14 +1,27 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { Store } from '@ngrx/store';
 import * as fromRoot from './reducers';
+import { Subscription } from 'rxjs/Subscription';
+import { Router, NavigationEnd } from '@angular/router';
 
 @Component({
 	selector: 'app-root',
 	templateUrl: './app.component.html',
 	styleUrls: ['./app.component.scss']
 })
-export class AppComponent {
+export class AppComponent implements OnInit {
 	constructor (
-		private store: Store<fromRoot.State>
+		private store: Store<fromRoot.State> ,
+		private router: Router
 	) { }
+
+	ngOnInit() {
+		let subscription: Subscription = this.router.events
```

Comment:
We should use the approach used in all the other components for handling subscriptions using `__subscriptions__ ` array. take hints from HomeComponent and FeedComponent. It Will be a better way to clearup all the subscriptions `OnDestroy`